inc/Engine/Preload/Database/Queries/Cache.php

```diff
@@ -122,7 +124,19 @@ public function __construct( Logger $logger, $query = [] ) {
 	 * @return bool
 	 */
 	public function create_or_update( array $resource ) {
-		$url = untrailingslashit( strtok( $resource['url'], '?' ) );
+		$url = $resource['url'];
+		/**
+		 * Filters to allow query string in preload.
+		 *
+		 * @param array $is_allowed Are query strings allowed.
+		 */
+		if ( apply_filters( 'rocket_preload_query_string', false ) ) {
```

Comment:
this filter is applied multiple times here in this file and in `inc/Engine/Preload/Controller/PreloadUrl.php` file, we need to unify it in one place

Reply:
@engahmeds3ed the filter is here to uniform the logic and prevent rewriting it multiple times, that's not that approach we should use  with filters?

Reply:
What I mean, we can have one method that contains the `apply_filters` and we can use this method whenever we want, think of it if u need to add a new argument to the filter or at least change its documentation, u will do it many times, what do u think?